File: packages/toolkit/src/padding.ts
const BLOCK = 4096;

export function pad(str: string): string {
  const remainder = str.length % BLOCK;
  if (remainder === 0) return str;
  const needed = BLOCK - remainder;
  const trimmed = str.replace(/\}\s*$/, '');
  return trimmed + ',' + ' '.repeat(needed - 1) + '}';
}

export function unpad(padded: string): string {
  return padded.replace(/,\s*\}$/, '}');
}

File: crates/msgpunk-crypto/src/padding.rs
const PADDING_BLOCK: usize = 4096;

pub fn pad(json_str: &str) -> String {
    let base_len = json_str.len();
    let remainder = base_len % PADDING_BLOCK;
    if remainder == 0 {
        return json_str.to_string();
    }
    let padding_needed = PADDING_BLOCK - remainder;
    let trimmed = json_str.trim_end_matches('}');
    let mut padded = trimmed.to_string();
    padded.push(',');
    padded.push_str(&" ".repeat(padding_needed - 1));
    padded.push('}');
    padded
}

pub fn unpad(padded_str: &str) -> String {
    let without_brace = padded_str.trim_end_matches('}');
    let trimmed = without_brace.trim_end_matches([' ', ',']);
    trimmed.to_string() + "}"
}
